mocks/tests/test_mocks.py

```diff
@@ -88,6 +88,22 @@ def test_scale_light_by_distance(particle_positions_3D,
                                                    1e-6)
 
 
+def test_scale_light_by_distance_float32(particle_positions_3D_float32,
+                                         mass_weights):
+    positions, luminosity =\
+        mocks.scale_light_by_distance(particle_positions_3D_float32,
+                                      mass_weights,
+                                      10,
+                                      4 * u.Mpc)
+    assert type(positions) == np.ndarray
+    assert type(luminosity) == u.Quantity
+    assert positions.shape == (5, 3)
+    assert luminosity.shape == (5,)
+    assert positions.max() == pytest.approx(18.999981407378996, rel=1e-12)
```

Comment:
Here it can be seen that the output of the function `scale_light_by_distance` for the same values but different dtypes is different.